pkg/acceptor/quic_acceptor.go

```diff
@@ -0,0 +1,244 @@
+func (a *QuicAcceptor) GetConnChan() chan PlayerConn {
+	return a.connChan
+}
+
+// Listen starts the QUIC listener to accept new connections
+func (a *QuicAcceptor) Listen() error {
+	// Creating a QUIC listener at the specified address
+	a.running = true
+	listener, err := quic.ListenAddr(a.addr, a.tlsConf, a.quicConf)
+	if err != nil {
+		return err
+	}
+	a.listener = listener
+	return nil
+}
+
+// Accept accepts new QUIC connections
+func (a *QuicAcceptor) Accept() (quic.Connection, error) {
+	if a.listener == nil { // Correct comparison
+		return nil, ErrListenerNotInitialized
+	}
+
+	conn, err := a.listener.Accept(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+// Close closes the QUIC listener
+func (a *QuicAcceptor) Close() error {
+	if a.listener != nil { // Correct comparison
+		return a.listener.Close()
+	}
+	return nil
+}
+
+// EnableProxyProtocol not implemented for QUIC, keep as No-op or implement as needed
+func (a *QuicAcceptor) EnableProxyProtocol() {
+	// No-op: Implement this method if needed for Proxy Protocol support
+}
+
+func (a *QuicAcceptor) IsRunning() bool {
+	return a.running
+}
+
+func (a *QuicAcceptor) GetConfiguredAddress() string {
+	return a.addr
+}
+
+func (a *QuicAcceptor) ListenAndServe() {
+	// Start the QUIC listener
+	if err := a.Listen(); err != nil {
+		fmt.Printf("Failed to start QUIC listener: %s\n", err)
+		return
+	}
+
+	// Loop to continuously accept connections
+	for a.IsRunning() {
+		conn, err := a.Accept()
+		if err != nil {
+			if errors.Is(err, ErrListenerNotInitialized) {
+				fmt.Println("Listener not initialized")
+				continue
+			}
+			fmt.Printf("Failed to accept connection: %s\n", err)
+			continue
+		}
+
+		// Send the connection to the channel for further processing
+		go func(c quic.Connection) {
+			playerConn := NewQuicConnWrapper(c)
+			a.connChan <- playerConn
+		}(conn)
+	}
+}
+
+// QuicConnWrapper is a wrapper for a QUIC connection, allowing the use of deadlines
+type QuicConnWrapper struct {
+	conn quic.Connection
+}
+
+// NewQuicConnWrapper creates a new wrapper for a QUIC connection
+func NewQuicConnWrapper(conn quic.Connection) *QuicConnWrapper {
+	return &QuicConnWrapper{conn: conn}
+}
+
+// Read reads data from the QUIC connection
+func (q *QuicConnWrapper) Read(p []byte) (int, error) {
+	stream, err := q.conn.AcceptStream(context.Background())
+	if err != nil {
+		return 0, err
+	}
+
+	return stream.Read(p)
+}
+
+// Write writes data to the connection with a defined deadline
+func (q *QuicConnWrapper) Write(p []byte) (int, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second) // 5 seconds timeout as an example
+	defer cancel()
+
+	stream, err := q.conn.OpenStreamSync(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return stream.Write(p)
+}
+
+// Close closes the QUIC connection
+func (q *QuicConnWrapper) Close() error {
+	return q.conn.CloseWithError(0, "closed")
```

Comment:
Didn't really read the API here, but do all connections need to be closed with errors?